exercises/03-xpath-intro.md

```diff
@@ -7,117 +7,87 @@
-open our developer console by right-clicking anywhere on the screen and selecting `inspect`.
-It's a good idea to keep the developer console always opened when you're writing Selenium tests.
-We notice, that the dropdown doesn't have an `id` field that would allow us to
-easily access that element.
-
-Let's start by observing the page's elements and their attributes. We notice that the dropdown button is an
-`a` element, which has classes we could use, for example `dropdown-toggle`. However, there's a similar, but hidden `a`
-element before in the HTML, so we can't use that `a` alone. Instead, we can use its parent
-`div` element to handle the click, as it is the size of the button. Also, it has a class called `open` when the dropdown is opened which disappears when it's closed. So, in other words _if_ the `div` element has a class called `open`, we can click it to close it.
-
-Still, the locator we use may look a bit obscure. Instead of adding it directly into our keyword, let's
-make a variable in the resource file. Following Robot Framework's best practices, we should give our
-variable an UPPER CASE name. As we write more code, we can add next XPaths into the table of `Variables`, every time giving them meaningful names.
-
-- Add `//div[contains(@class, 'open')]` into a variable with a meaningful name, such
-as `OPENED DROPDOWN`.
-- Create a keyword called `Close Dropdown If Opened` that clicks the element `OPENED DROPDOWN`.
+- Using your browser's console, look for a suitable locator to identify an opened dropdown.
+- Create a variable in Variables table, called `OPENED DROPDOWN` and give it the value of your locator.
+- Add a new keyword called `Close Dropdown If Opened`.
+   - Check if the dropdown is opened with a library-specific keyword.
+   - Catch the result of that keyword into a local variable, called `${element visible}`.
+   - You can use a keyword from the BuiltIn library to click the dropdown if `${element visible}` is true.
 
+### Useful keywords
 <details>
   <summary>SeleniumLibrary</summary>
 
-To check if the element is visible we need to first get the element status, then combine that with
-`Run Keyword If`. We can ge the element status with `Run Keyword And Return Status` combined with
-`Page Should Contain Element`
-
-- Use `Run Keyword And Return Status` and `Page Should Contain Element` to check if the dropdown is opened.
-Store the result in a variable.
-- Using your new variable, use `Run Keyword If` to conditionally close the dropdown.
+- Use BuiltIn library's `Run Keyword And Return Status` and SeleniumLibrary's `Page Should Contain Element` to check if the dropdown is opened.
 
 </details> <!-- SeleniumLibrary -->
 
 <details>
   <summary>Browser</summary>
 
-To check if the element is visible, we can use `Get Element State` and check the state for `visible`.
-
 - Use `Get Element State` with `visible` as the state and store it in a variable.
```

Comment:
`Get Element State` is actually deprecated, but the whole exercise needs a little work to address that. I'll make a note of that and let's handle this in another PR.